### ISSUE TITLE
Bug: id field is always an ID scalar type, even when the actual type in the schema is an Int

### ISSUE THREAD

**Issue by capaj** [YassinEldeeb/prisma-tgql-types-gen#37]
## Bug description
I have a schema like this:
```
model NastyModelWithIntId {
  id Int @id @default(autoincrement())
}
```

output from the generator is:

```ts
import { Field, ID, ObjectType } from 'type-graphql'

@ObjectType()
export class NastyModelWithIntIdScalars {
  @Field((_type) => ID)
  id: number
}

export class NastyModelWithIntId extends NastyModelWithIntIdScalars {
  // skip overwrite 👇
}

```
Due to this function:
![image](https://user-images.githubusercontent.com/1305378/147894043-ff051e2d-3dc8-423e-a039-808d89a250d8.png)

## To Reproduce
just generate

## Expected behavior
The output should be like this:
```ts
import { Field, ID, ObjectType } from 'type-graphql'

@ObjectType()
export class NastyModelWithIntIdScalars {
  @Field((_type) => Int)
  id: number
}

export class NastyModelWithIntId extends NastyModelWithIntIdScalars {
  // skip overwrite 👇
}

```

## Prisma  Version
any

**Comment by YassinEldeeb** [comment]
> the id field gets generated like this:
> ```
> @Field(() => ID)
> id: number
> ```
> The output should be like this:
>```
> @Field(() => GraphqlInt)
> id: number
>```

what do you mean? `Int` is an alias for `GraphqlInt` in Typegraphql.

so what's the point of switching them if you are using Typegraphql.

**Comment by YassinEldeeb** [comment]
Also another note:
> the id field gets generated like this:
> ```
> @Field(() => ID)
> id: number
> ```

It doesn't get generated like that, the ID is always a string so It's generated like this:
```ts
@Field(() => ID)
id: string
```

**Comment by YassinEldeeb** [comment]
I don't understand the bug, the title says something and the expected behavior is talking about something else.

**Comment by capaj** [comment]
When you have a model which has Int as and `id` it doesn't sit well with graphql itself. because in graphql spec `ID` scalar is always a string.
IMHO it is better to use an `Int` instead of `ID`, because an Int works even if it get's returned as a string doesn't provide the uniquness guarantees that some GQL clients expect from an `ID`.

Basically and `ID` should always be a UUID, or at least something pseudo unique across the whole API, so for example a string like `user_12343`. When we generate and there is an Int there, it should have an Int scalar on the GQL side too.

**Comment by capaj** [comment]
>It doesn't get generated like that, the ID is always a string so It's generated like this:

it does not. Try it. id will be marked as `number` and as `ID` in the decorator.

**Comment by YassinEldeeb** [comment]
> > It doesn't get generated like that, the ID is always a string so It's generated like this:
> 
> it does not. Try it. id will be marked as `number` and as `ID` in the decorator.

I meant what my generator is outputting not IDs stored in databases in general.

**Comment by YassinEldeeb** [comment]
> When you have a model which has Int as and `id` it doesn't sit well with graphql itself. because in graphql spec `ID` scalar is always a string. IMHO it is better to use an `Int` instead of `ID`, because an Int works even if it get's returned as a string doesn't provide the uniquness guarantees that some GQL clients expect from an `ID`.
> 
> Basically and `ID` should always be a UUID, or at least something pseudo unique across the whole API, so for example a string like `user_12343`. When we generate and there is an Int there, it should have an Int scalar on the GQL side too.

Yeah convincing, you're right. The type is more accurate to be an `Int` in this case instead of a `ID` which gets transformed to `GraphQLID`

**Comment by YassinEldeeb** [comment]
sorry, I've misunderstood you completely at first.

**Comment by capaj** [comment]
no worries-the title was a bit misleading too :laughing:

**Comment by YassinEldeeb** [comment]
It's fixed now with `v1.5.11` 🎉
